webcord: 4.2.0 -> 4.3.0, add updateScript

huantianad opened the pull request:
###### Description of changes
Update to latest minor version, includes an electron version bump. Adds a default `nix-update-script` as `passthru.updateScript`
<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

huantianad committed:
webcord: 4.2.0 -> 4.3.0, add updateScript

FlafyDev commented:
Webcord from this PR works for me.

But the switch to electron 25 makes `webcord-vencord` crash(segfault) on launch in Wayland.
Should we override electron_25 to electron_24 for the `webcord-vencord` package?

huantianad commented:
Hm yeah if this webcord package works with electron 25 on wayland, but not webcord-vencord, then overriding it might be the best option. Should I include that in this PR?

FlafyDev commented:
> Hm yeah if this webcord package works with electron 25 on wayland, but not webcord-vencord, then overriding it might be the best option. Should I include that in this PR?

I'm not sure whether to include the override in this PR. But I guess it makes sense to include it, otherwise webcord-vencord breaks until another PR is made and merged, which I believe is less preferable.

huantianad commented:
I did notice that there are recent updates to vencord itself, could those fix the issues on electron 25 as well? If no luck, I'll add a commit to override the electron version.

FlafyDev commented:
I tried with [vencord's latest commit](https://github.com/Vendicated/Vencord/commit/c696c186e82bd84c665ae5ba0dfc2a89863b435a) and still segfault.
the error is probably not caused by the extension itself

huantianad committed:
webcord-vencord: override electron to use electron_24

Webcord has updated to electron 25, but that causes a segfault when launching webcord-vencord on wayland, so downgrade it for now.

Co-authored-by: FlafyDev <flafyarazi@gmail.com>

huantianad commented:
Ok just committed the fix then, lemme know if it works!

FlafyDev commented on pkgs/applications/networking/instant-messengers/webcord/webcord-vencord/default.nix:
```diff
@@ -11,6 +12,10 @@ webcord.overrideAttrs (old: {
 
+  # Webcord has updated to electron 25, but that causes a segfault
+  # when launching webcord-vencord on wayland, so downgrade it for now.
+  electron_25 = electron_24;
```
This doesn't actually do anything since `overrideAttrs` is for overriding attributes passed to mkDerivation.

To override electron_25 we need to use `override`

I don't know if there's a better way to do this, but that works:

```nix
{ webcord
, substituteAll
, lib
, vencord-web-extension
, electron_24
,
}:
(webcord.overrideAttrs (old: {
  patches = (old.patches or [ ]) ++ [
    (substituteAll {
      src = ./add-extension.patch;
      vencord = vencord-web-extension;
    })
  ];

  meta = with lib;
    old.meta
    // {
      description = "Webcord with Vencord web extension";
      maintainers = with maintainers; [ FlafyDev NotAShelf ];
    };
})).override {
  # Webcord has updated to electron 25, but that causes a segfault
  # when launching webcord-vencord on wayland, so downgrade it for now.
  electron_25 = electron_24;
}
```

huantianad replied:
wups I always make this mistake will fix

FlafyDev reviewed (approved):
LGTM

Gerg-L commented on pkgs/applications/networking/instant-messengers/webcord/default.nix:
```diff
@@ -1,18 +1,29 @@
-{ lib, stdenv, buildNpmPackage, fetchFromGitHub, copyDesktopItems
-, python3, pipewire, libpulseaudio, xdg-utils, electron_24, makeDesktopItem }:
+{ lib
+, stdenv
```
`, stdenv` isn't used please remove

Gerg-L commented on pkgs/applications/networking/instant-messengers/webcord/default.nix:
```diff
@@ -1,18 +1,29 @@
     repo = "WebCord";
     rev = "v${version}";
-    sha256 = "sha256-530iWNvehImwSYt5HnZaqa4TAslrwxAOZi3gRm1K2/w=";
+    sha256 = "sha256-E/WXAVSCNTDEDaz71LXOHUf/APFO2uSpkTRhlZfQp0E=";
```
use `hash =` instead of `sha256 =`

Gerg-L reviewed (approved):
Thank you for maintaining WebCord, I use it daily LGTM!

adamcstephens reviewed (approved):
x86_64-linux - built and ran successfully
aarch64-linux - built

`meta.platforms` should probably be set to Linux only, or the pipewire dependencies guarded to only Linux. As-is pipewire prevents Darwin from building.

huantianad commented:
Ah, if it fails to compile on Darwin I'll probably remove it from `meta.platforms` as I have no way of maintaining the package for darwin myself, unless anyone else wants to help maintain for darwin.
